src/widgets/Header/InterfaceSettings.tsx

```diff
@@ -9,14 +10,21 @@ export default function InterfaceSettings() {
   const { themeValue, setThemeValue } = useContext(ThemeContext);
   const router = useRouter();
   const pathname = usePathname();
+  const locale = useLocale();
 
   const handleThemeChange = (checked: boolean) => {
     setThemeValue(checked ? 'dark' : 'light');
   };
 
+  const switchLocale = (newLocale: string) => {
+    if (newLocale !== locale) {
+      router.replace(pathname, { locale: newLocale });
+      router.refresh();
+    }
+  };
+
```

Comment:
I think the removal seems unnecessary.
I would leave it as it is.

```suggestion
  const handleMenuClick: MenuProps['onClick'] = (e) => {
    const newLocale = e.key
  
    if (newLocale !== locale) {
      router.replace(pathname, { locale: newLocale });
      router.refresh();
    }
  };

```

Reply:
Removed the switchLocale and moved the logic into handleMenuClick.